flink-runtime/src/main/java/org/apache/flink/runtime/state/StateBackendLoader.java

```diff
@@ -131,7 +144,7 @@ public static StateBackend loadStateBackendFromConfig(
 
             case ROCKSDB_STATE_BACKEND_NAME:
                 factoryClassName =
-                        "org.apache.flink.contrib.streaming.state.RocksDBStateBackendFactory";
+                        "org.apache.flink.contrib.streaming.state.EmbeddedRocksDBStateBackendFactory";
```

Comment:
I think this change breaks backwards combatibility (with old configs). 
Should we fallback to the older factory/name?

Reply:
No, it doesn't. If you do not explicitly specify a checkpoint storage policy and a checkpoint directory is provided it will use FileSystemCheckpointStorage which is equivalent to the old RocksDBStateBackend.

Reply:
You're right, thanks. I misread this part of code.